### PR TITLE
Adapt step function

### PR DESCRIPTION
**Pull request by Coderilz**

### CODE REVIEW

File: architecture/step-function/main.tf [hunk 1]
```diff
@@ -119,11 +119,11 @@ resource "aws_iam_role_policy_attachment" "function_logging_policy_attachment" {
   policy_arn = aws_iam_policy.function_logging_policy.arn
 }
 
-# lamdba extract stuff
+# lamdba extract spotify stuff
```
Comment (Selvy96) [inline architecture/step-function/main.tf:122]: Maybe we could remove the word 'stuff'

File: architecture/ecr/update_tiktok/update.tf [hunk 2]
```diff
@@ -14,7 +14,7 @@ provider "aws" {
 }
 
 resource "aws_ecr_repository" "long_term_image" {
-  name                 = "spotify-tiktok-1-hour-update"
+  name                 = "spotify-tiktok-update-tiktok"
```
Comment (KejviBode) [inline architecture/ecr/update_tiktok/update.tf:17]: Maybe a better variable name for the ecr could make it more clear what the image is doing

File: architecture/step-function/main.tf [hunk 3]
```diff
@@ -190,13 +215,15 @@ resource "aws_lambda_function" "spotify-tiktok-report" {
       DB_PASSWORD   = var.DB_PASSWORD
       ACCESS_KEY_ID = var.ACCESS_KEY_ID
       SECRET_KEY_ID = var.SECRET_KEY_ID
+      CLIENT_ID     = var.CLIENT_ID
+      CLIENT_SECRET = var.CLIENT_SECRET
     }
   }
 }
 
 # sns topic stuff
```
Comment (Selvy96) [inline architecture/step-function/main.tf:224]: And here too maybe

File: architecture/step-function/main.tf [hunk 4]
```diff
@@ -214,6 +241,54 @@ resource "aws_sns_topic_subscription" "selvy-target" {
...
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "sns_publish_attachment" {
+  policy_arn = aws_iam_policy.sns_publish_policy.arn
+  role       = aws_iam_role.step_function_role.name
+}
+
 
 # step-function stuff
```
Comment (Selvy96) [inline architecture/step-function/main.tf:293]: Also here

**Review by Selvy96 — APPROVED** [review]

**Review by Selvy96 — APPROVED** [review]

**Pull request merged**